### PR TITLE
Feature/426947 get submissions data

### PR DESCRIPTION
**Pull request by jayakrishnanappusamy**
Added store approved submission function

### CODE REVIEW

**Review by harishpaladugu13 — DISMISSED** [review]

File: src/EPR.PRN.ObligationCalculation.Application/Configs/AppInsightsConfig.cs
Comment (saywasingh-eviden) [inline src/EPR.PRN.ObligationCalculation.Application/Configs/AppInsightsConfig.cs:1]: Filename is not appropriate

File: src/EPR.PRN.ObligationCalculation.Function/StoreApprovedSubmissionsFunction.cs [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
+        public async Task RunAsync([TimerTrigger("%StoreApprovedSubmissions:Schedule%")] TimerInfo myTimer)
         {
-            _logger.LogInformation($"{LogPrefix} New session started");
+            _logger.LogInformation("{LogPrefix} >>>>> New session started <<<<< ", LogPrefix);
+
+            var createdDate = _appInsightsProvider.GetParameterForApprovedSubmissionsApiCall().Result;
+            var approvedSubmissionEntities = await _submissionsService.GetApprovedSubmissionsData(createdDate.ToString("yyyy-MM-dd"));
 
-            if (myTimer.ScheduleStatus is not null)
+            if (approvedSubmissionEntities.Count > 0)
```
Comment (saywasingh-eviden) [inline src/EPR.PRN.ObligationCalculation.Function/StoreApprovedSubmissionsFunction.cs:34]: Can this inverted to something like

if (approved... .Count == 0 )
{
_logger.LogInformation("{LogPrefix} COMPLETED >>>>> No new submissions received and send to queue <<<<< ", LogPrefix);
return;
}

File: src/EPR.PRN.ObligationCalculation.Application/Services/SubmissionsDataService.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        try
+        {
+            var result = await GetDataAsync(endpoint);
+
+            if (string.IsNullOrEmpty(result))
+            {
+                _logger.LogWarning("{LogPrefix} No submissions data found for {ApprovedAfterDateString}", ApplicationConstants.StoreApprovedSubmissionsFunctionLogPrefix, approvedAfterDateString);
+                return new List<ApprovedSubmissionEntity>();
+            }
+            else
```
Comment (saywasingh-eviden) [inline src/EPR.PRN.ObligationCalculation.Application/Services/SubmissionsDataService.cs:45]: The "else" is not needed.

File: src/EPR.PRN.ObligationCalculation.Application/Services/ServiceBusProvider.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        private readonly ILogger<ServiceBusProvider> _logger;
+        private readonly ServiceBusClient _serviceBusClient;
+        private readonly ServiceBusConfig _config;
+
+        public ServiceBusProvider(ILogger<ServiceBusProvider> logger, ServiceBusClient serviceBusClient, IOptions<ServiceBusConfig> config)
+        {
+            _logger = logger;
+            _serviceBusClient = serviceBusClient;
+            _config = config.Value;
+        }
```
Comment (saywasingh-eviden) [inline src/EPR.PRN.ObligationCalculation.Application/Services/ServiceBusProvider.cs:23]: An extra line break needed here

File: src/EPR.PRN.ObligationCalculation.Application/Services/SubmissionsDataService.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+using Newtonsoft.Json;
+using System.Net;
+
+namespace EPR.PRN.ObligationCalculation.Application.Services;
+
+public class SubmissionsDataService : ISubmissionsDataService
+{
+    private readonly ILogger<SubmissionsDataService> _logger;
+    private readonly HttpClient _httpClient;
+    private readonly SubmissionsApiConfig _config;
```
Comment (saywasingh-eviden) [inline src/EPR.PRN.ObligationCalculation.Application/Services/SubmissionsDataService.cs:14]: Extra line break needed here.

File: src/EPR.PRN.ObligationCalculation.Application/Services/AppInsightsProvider.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+using Microsoft.Extensions.Options;
+
+namespace EPR.PRN.ObligationCalculation.Application.Services
+{
+    public class AppInsightsProvider : IAppInsightsProvider
+    {
+        private readonly ILogger<AppInsightsProvider> _logger;
+        private readonly LogsQueryClient _logsQueryClient;
+        private readonly AppInsightsConfig _config;
+        private const string TimeGenerated = "TimeGenerated";
```
Comment (saywasingh-eviden) [inline src/EPR.PRN.ObligationCalculation.Application/Services/AppInsightsProvider.cs:16]: Would this be better placed inside the Constants class? What do you guys think?

**Pull request closed without merging**